rust/src/rendezvous/mod.rs

```diff
@@ -242,11 +245,17 @@ impl RendezvousHandler {
             let mut headers = HeaderMap::new();
             prepare_headers(&mut headers, session);
 
+            let mut additional_fields = HashMap::with_capacity(1);
+            additional_fields.insert(
+                String::from("org.matrix.msc4108.errcode"),
+                String::from("M_CONCURRENT_WRITE"),
+            );
+
             return Err(SynapseError::new(
                 StatusCode::PRECONDITION_FAILED,
                 "ETag does not match".to_owned(),
-                "M_CONCURRENT_WRITE",
-                None,
+                "M_UNKNOWN", // Would be M_CONCURRENT_WRITE
+                Option::from(additional_fields),
```

Comment:
It's more common to write `Some(additional_fields)` instead.

```suggestion
                Some(additional_fields),
```